internal/config/variables/variables.go

```diff
@@ -0,0 +1,531 @@
+	Source string
+	Expr   hcl.Expression
+	// The location of the variable value if the value was provided from a file
+	Range hcl.Range
+}
+
+// DecodeVariableBlock validates each part of the variable block,
+// building out a defined *Variable
+func DecodeVariableBlock(block *hcl.Block) (*Variable, hcl.Diagnostics) {
+	name := block.Labels[0]
+	v := Variable{
+		Name:  name,
+		Range: block.DefRange,
+	}
+
+	content, diags := block.Body.Content(variableBlockSchema)
+
+	if !hclsyntax.ValidIdentifier(name) {
+		diags = append(diags, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid variable name",
+			Detail:   "A name must start with a letter or underscore and may contain only letters, digits, underscores, and dashes.",
+			Subject:  &block.LabelRanges[0],
+		})
+	}
+
+	if attr, exists := content.Attributes["description"]; exists {
+		valDiags := gohcl.DecodeExpression(attr.Expr, nil, &v.Description)
+		diags = append(diags, valDiags...)
+		if diags.HasErrors() {
+			return nil, diags
+		}
+	}
+
+	if attr, ok := content.Attributes["type"]; ok {
+		t, moreDiags := typeexpr.Type(attr.Expr)
+		diags = append(diags, moreDiags...)
+		if moreDiags.HasErrors() {
+			return nil, diags
+		}
+		v.Type = t
+	}
+
+	if attr, exists := content.Attributes["default"]; exists {
+		val, valDiags := attr.Expr.Value(nil)
+		diags = append(diags, valDiags...)
+		if diags.HasErrors() {
+			return nil, diags
+		}
+		// Convert the default to the expected type so we can catch invalid
+		// defaults early and allow later code to assume validity.
+		// Note that this depends on us having already processed any "type"
+		// attribute above.
+		if v.Type != cty.NilType {
+			var err error
+			val, err = convert.Convert(val, v.Type)
+			if err != nil {
+				diags = append(diags, &hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  fmt.Sprintf("Invalid default value for variable %q", name),
+					Detail:   fmt.Sprintf("This default value is not compatible with the variable's type constraint: %s.", err),
+					Subject:  attr.Expr.Range().Ptr(),
+				})
+				val = cty.DynamicVal
+			}
+		}
+
+		v.Default = &InputValue{
+			Source: sourceDefault,
+			Value:  val,
+		}
+
+		// It's possible no type attribute was assigned so lets make sure we
+		// have a valid type otherwise there could be issues parsing the value.
+		if v.Type == cty.NilType {
+			v.Type = val.Type()
+		}
+	}
+
+	return &v, diags
+}
+
+// SetJobInputValues collects values set via the CLI (-var, -varfile) and
+// local env vars (WP_VAR_*) and translates those values to pb.Variables. These
+// pb.Variables can then be set on the job for eventual parsing on the runner,
+// after the runner has decoded the variables defined in the waypoint.hcl.
+// All values are set as protobuf strings, with the expectation that later
+// evaluation will convert them to their defined types.
+func SetJobInputValues(vars map[string]string, files []string) ([]*pb.Variable, hcl.Diagnostics) {
```

Comment:
`Set` prefix here makes it sound like this is setting some state. It made me scared you were using a global var somewhere! But that isn't the case. 

I think it'd be more idiomatic to simply drop the `Set`. 

It is also confusing because you actually have a thing called `InputValues` and this seemingly has nothing to do with it; it returns variables. So maybe this would make more sense to be named `LoadVariableValues` or something.

Reply:
Hmmm yeah I think that makes sense. The returned set of pb.Variables could theoretically be used for anything, not specifically the job, so that doesn't need to be in the func name. `LoadVariableValues` is as good as anything.